src/SurfacePlot/SurfacePlot.tsx

```diff
@@ -0,0 +1,202 @@
+import {
+  Box,
+  Dialog,
+  DialogContent,
+  Typography,
+  useTheme
+} from "@mui/material";
+import React, { useState } from "react";
+
+import DialogTitle from "../DialogTitle";
+import Plotly from "react-plotly.js";
+import { SurfacePlotProps } from "./SurfacePlot.types";
+
+// Define the SurfacePlot component
```

Comment:
need a better comment here - these first comments at the top should give a brief description of what the component is used for